Zhongli.Data/Models/Logging/UserJoinLog.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Zhongli.Data.Models.Discord;
+
+namespace Zhongli.Data.Models.Logging;
+
+public class UserJoinLog : ILog, IGuildUserEntity
+{
+    public UserJoinLog(GuildUserEntity user, DateTimeOffset? userJoinDate)
+    {
+        LogDate       = DateTimeOffset.UtcNow;
+        User          = user;
+        UserId        = user.Id;
+        GuildId       = user.GuildId;
+        FirstJoinDate = user.JoinedAt;
+        JoinDate      = userJoinDate;
```

Comment:
The `JoinDate` can be inferred from the `LogDate` so the only important thing is the `FirstJoinDate`